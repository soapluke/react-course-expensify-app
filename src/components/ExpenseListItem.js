import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}><h3>Description: {description}</h3></Link>
        <p>Amount: {amount}</p>
        <p>Created At: {moment(createdAt).format('MMMM Do, YYYY')}</p>
    </div>
);

export default ExpenseListItem;