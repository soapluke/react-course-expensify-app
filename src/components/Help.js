import React from 'react';

const Help = () => (
    <div>
        This is my help component.
    </div>
);

export default Help;