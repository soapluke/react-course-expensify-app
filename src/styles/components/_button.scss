.button {
    background-color: $blue;
    color: white;
    display: inline-block;
    text-decoration: none;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    border: none;
}

.button--secondary {
    background-color: #888888;
}

.button--link {
    background: none;
}