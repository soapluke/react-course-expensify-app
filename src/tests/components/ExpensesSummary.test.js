import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={25346335}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('', () => {

});