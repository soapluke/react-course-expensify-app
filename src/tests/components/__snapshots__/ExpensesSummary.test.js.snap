// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    235
    .
  </h3>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h3>
    Viewing 
    23
     
    expenses
     totalling 
    25346335
    .
  </h3>
</div>
`;
